# 10term.patch by Niko Tyni <ntyni@iki.fi>
# DP: Fix uninitialized value message 
# DP: (#99843, patch by Nicolas Bertolissio)
--- a/Gnu/XS.pm
+++ b/Gnu/XS.pm
@@ -363,11 +363,12 @@
 use vars qw(%term_no_ue);
 %term_no_ue = ( kterm => 1 );
 
+
 sub ornaments {
     return $rl_term_set unless @_;
     $rl_term_set = shift;
     $rl_term_set ||= ',,,';
-    $rl_term_set = $term_no_ue{$ENV{TERM}} ? 'us,me,,' : 'us,ue,,'
+    $rl_term_set = $term_no_ue{defined($ENV{TERM}) ? $ENV{TERM} : ''} ? 'us,me,,' : 'us,ue,,'
 	if $rl_term_set eq '1';
     my @ts = split /,/, $rl_term_set, 4;
     my @rl_term_set
